lib/perl/Genome/Model/Tools/Transcriptome/ErccMapUnaligned.pm

```diff
@@ -0,0 +1,367 @@
+package Genome::Model::Tools::Transcriptome::ErccMapUnaligned;
+
+use strict;
+use warnings;
+
+use Genome;
+use Path::Class;
+
+class Genome::Model::Tools::Transcriptome::ErccMapUnaligned {
+    is => 'Command::V2',
+    has_input => [
+        bam_file => {
+            is => 'Text',
+            doc => 'An aligned BAM file to a reference genome without the ERCC transcripts spiked in.',
+        },
+        ercc_fasta_file => {
+            is => 'Text',
+            doc => 'The FASTA format sequence for all ERCC transcripts.',
+            example_values => ['/gscmnt/gc13001/info/model_data/jwalker_scratch/ERCC/ERCC92.fa'],
+        },
+        ercc_spike_in_file => {
+            is => 'Text',
+            doc => 'The control analysis file provided by Agilent for the ERCC spike-in.',
+            example_values => ['/gscmnt/gc13001/info/model_data/jwalker_scratch/ERCC/ERCC_Controls_Analysis.txt'],
+        },
+        ercc_spike_in_mix => {
+            is => 'Integer',
+            doc => 'The expected ERCC spike-in mix.',
+            valid_values => [ '1', '2'],
+        },
+        samtools_version => {
+            is => 'Text',
+            doc => 'The version of samtools to run analysis.',
+            example_values => ['1.2'],
+            default_value => '1.2',
+            is_optional => '1',
+        },
+        samtools_max_mem => {
+            is => 'Text',
+            doc => 'The max memory required by samtools.',
+            example_values => ['14000000000'],
+            default_value => '14000000000',
+            is_optional => '1',
+        },
+        bowtie2_version => {
+            is => 'Text',
+            doc => 'The version of bwa to run analysis.',
+            example_values => ['2.1.0'],
+            default_value => '2.1.0',
+            is_optional => '1',
+        },
+    ],
+    has_output => [
+        pdf_file => {
+            is => 'Text',
+            doc => 'The output PDF with histograms and linearity plot.',
+            default_value => 'output.pdf',
+            is_optional => '1',
+        }
+    ],
+};
+
+sub help_detail {
+    return <<EOS
+Compare the abundance of an ERCC spike-in with a known concentration.
+EOS
+}
+
+sub execute {
+    my $self = shift;
+    
+    my $ercc_bowtie_index = $self->create_ercc_bowtie_index();
+
+    my $remapped_bam = $self->generate_remapped_bam(
+        index => $ercc_bowtie_index,
+        input_bam => $self->bam_file,
+    );
+
+    $self->index_bam($remapped_bam);
+    my $idxstats = $self->generate_idxstats($remapped_bam);
+    my $tsv = $self->generate_tsvfile($idxstats);
+    $self->copy_to( $tsv, Path::Class::Dir->new() );
```

Comment:
What this script is going to be finally outputting is still being finalized.  Again deferred for a Part 2 pull request.